DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Centre/Reports/EditFiltersViewModel.cs

```diff
@@ -0,0 +1,167 @@
+        {
+            var validationResults = new List<ValidationResult>();
+
+            var startDateValidationResults = DateValidator.ValidateDate(
+                    StartDay,
+                    StartMonth,
+                    StartYear,
+                    "Start date",
+                    true,
+                    false,
+                    true
+                )
+                .ToValidationResultList(nameof(StartDay), nameof(StartMonth), nameof(StartYear));
+
+            validationResults.AddRange(startDateValidationResults);
+
+            if (EndDate)
+            {
+                var endDateValidationResults = DateValidator.ValidateDate(
+                        EndDay,
+                        EndMonth,
+                        EndYear,
+                        "End date",
+                        true,
+                        false,
+                        true
+                    )
+                    .ToValidationResultList(nameof(EndDay), nameof(EndMonth), nameof(EndYear));
+
+                if (StartYear > EndYear
+                    || StartYear == EndYear && StartMonth > EndMonth
+                    || StartYear == EndYear && StartMonth == EndMonth && StartDay > EndDay)
+                {
+                    endDateValidationResults.Add(
+                        new ValidationResult(
+                            "End date must not precede start date",
+                            new[]
+                            {
+                                nameof(StartDay)
+                            }
+                        )
+                    );
+                    endDateValidationResults.Add(
+                        new ValidationResult(
+                            "",
+                            new[]
+                            {
+                                nameof(StartMonth), nameof(StartYear), nameof(EndDay),
+                                nameof(EndMonth), nameof(EndYear)
+                            }
+                        )
+                    );
+                }
+
+                validationResults.AddRange(endDateValidationResults);
+            }
+
+            return validationResults;
+        }
+
+        public void SetUpDropdownOptions(
+            IEnumerable<(int, string)> jobGroupOptions,
+            IEnumerable<(int, string)> courseCategoryOptions,
+            IEnumerable<(int, string)> courseOptions,
+            int userCategoryId
+        )
+        {
+            CanFilterCourseCategories = userCategoryId == 0;
+
+            JobGroupOptions = SelectListHelper.MapOptionsToSelectListItems(jobGroupOptions, JobGroupId);
+            CourseCategoryOptions =
+                SelectListHelper.MapOptionsToSelectListItems(courseCategoryOptions, CourseCategoryId);
+            CustomisationOptions = SelectListHelper.MapOptionsToSelectListItems(courseOptions, CustomisationId);
+            var reportIntervals = Enum.GetValues(typeof(ReportInterval))

```

Comment:
I wonder if instead of this we should actually declare ReportInterval as an Enumeration class, which has an id and an explicit display name. 🤔